src/execution_plan/ops/op_node_by_id_seek.c

```diff
@@ -0,0 +1,97 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "op_node_by_id_seek.h"
+
+#define MIN(a,b) (((a) < (b)) ? (a) : (b))
+#define MAX(a,b) (((a) < (b)) ? (b) : (a))
+
+// Checks to see if operation index is within its bounds.
+static inline bool _outOfBounds(OpNodeByIdSeek *op) {
+    /* Because currentId starts at minimum and only increases
+     * we only care about top bound. */
+    if(op->currentId > op->maxId) return true;
+    if(op->currentId == op->maxId && !op->maxExclusive) return true;
+    return false;
+}
+
+OpBase* NewOpNodeByIdSeekOp
+(
+    const AST *ast,
+    unsigned int nodeRecIdx,
+    NodeID minId,
+    NodeID maxId,
+    bool minExclusive,
+    bool maxExclusive
+)
+{
+    // Can't include unspecified bound.
+    assert(!(minId == ID_RANGE_UNBOUND && minExclusive));
+    assert(!(maxId == ID_RANGE_UNBOUND && maxExclusive));
+
+    OpNodeByIdSeek *op_nodeByIdSeek = malloc(sizeof(OpNodeByIdSeek));
+    op_nodeByIdSeek->g = GraphContext_GetFromTLS()->g;
+    
+    op_nodeByIdSeek->minExclusive = minExclusive;
+    op_nodeByIdSeek->maxExclusive = maxExclusive;
+
+    // The smallest possible entity ID is 0.
+    op_nodeByIdSeek->minId = MAX(0, minId);
+    // The largest possible entity ID is the same as Graph_RequiredMatrixDim.
+    op_nodeByIdSeek->maxId = MIN(Graph_RequiredMatrixDim(op_nodeByIdSeek->g), maxId);
+
+    op_nodeByIdSeek->currentId = op_nodeByIdSeek->minId;
+    if(!minExclusive) op_nodeByIdSeek->currentId++;
```

Comment:
Given my comment at 42, something interesting happens here:
Given a clause like `WHERE ID(a) < 1`, `minID` is actually set to the largest possible value. This condition triggers, though, and `currentID` rolls over to 0.

After making the fix at 42, this condition needs to be `if(!minInclusive && minId != ID_RANGE_UNBOUND)`

Reply:
👍